### PR TITLE
Feature/sorted table

### PR DESCRIPTION
**Pull request by ivanzusko**

### CODE REVIEW

File: src/store/selectors/apiList.selector.js [hunk 1]
```diff
@@ -1,22 +1,49 @@
...
-    if (
-        el.name.toLowerCase().includes(searchQuery.toLowerCase()) ||
-        el.proxy.listen_path.toLowerCase().includes(searchQuery.toLowerCase()) ||
-        el.proxy.upstream_url.toLowerCase().includes(searchQuery.toLowerCase())
-    ) {
-        return el;
-    }
+const sortByNameCaseInsensitive = asc => R.sort(
+    asc
+        ? R.ascend(R.compose(R.toLower, R.prop('name')))
```
Comment (rafaeljesus) [inline src/store/selectors/apiList.selector.js]: 👌

File: src/store/selectors/apiList.selector.js [hunk 2]
```diff
@@ -1,22 +1,49 @@
...
+);
+const sortByActive = asc => R.sort(
+    asc
+        ? R.ascend(R.prop('active'))
+        : R.descend(R.prop('active'))
+);
+
+const getFilteredApiList = (apiList, searchQuery, sortingFilter, sortAscend) => {
+    const sortedList = (list, filterName, ascend) => {
+        if (filterName === 'name') {
```
Comment (rafaeljesus) [inline src/store/selectors/apiList.selector.js]:
```js
switch filterName {
  case 'name':
    return sortByNameCaseInsensitive(sortAscend)(list);
  case 'active':
    return sortByActive(sortAscend)(list);
  default:
    return list
}
```

File: src/store/selectors/apiList.selector.js [hunk 3]
```diff
@@ -1,22 +1,49 @@
...
+            return sortByNameCaseInsensitive(sortAscend)(list);
+        } else if (filterName === 'active') {
+            return sortByActive(sortAscend)(list);
+        }
+        return list;
+    };
+
+    return sortedList(apiList, sortingFilter).filter((el) => {
+        if (
```
Comment (rafaeljesus) [inline src/store/selectors/apiList.selector.js]: I 'd rather to have smh like
```js
return sortedList(apiList, sortingFilter).filter(el => {
  const hasSearch = el.name.toLowerCase().includes(searchQuery.toLowerCase())
    || el.proxy.listen_path.toLowerCase().includes(searchQuery.toLowerCase())
    || el.proxy.upstream_url.toLowerCase().includes(searchQuery.toLowerCase())
  
  if (hasSearch) return el;

  return false;
});
```

Also let's keep an eyes open for `R.compose` method

Comment (ivanzusko) [reply src/store/selectors/apiList.selector.js]: Smthng like this? :)
```js
return R.compose(listFilteredAccordingToSearchQuery, sortedList)(
        apiList,
        sortingFilter,
        sortAscend,
    );
```

File: src/store/selectors/apiList.selector.js [hunk 4]
```diff
@@ -1,22 +1,54 @@
...
+        const searchIsActive = el.name.toLowerCase().includes(searchQuery.toLowerCase()) ||
+            el.proxy.listen_path.toLowerCase().includes(searchQuery.toLowerCase()) ||
+            el.proxy.upstream_url.toLowerCase().includes(searchQuery.toLowerCase());
+
+        return (searchIsActive) ? el : false;
+    });
 
-    return false;
-});
+    return R.compose(listFilteredAccordingToSearchQuery, sortedList)(
```
Comment (rafaeljesus) [inline src/store/selectors/apiList.selector.js]: 😮 🏅

**Pull request merged**